evm/rlp/headers.py

```diff
@@ -179,83 +179,135 @@ def create_execution_context(self, prev_hashes):
+        chunks_root,
+        period,
+        proposer_address,
+        proposer_bid,
+    ):
+        """Initialize a new unsigned collation header as a child from another header."""
+        header_kwargs = {
+            "shard_id": parent.shard_id,
+            "parent_hash": parent.hash,
+            "chunks_root": chunks_root,
+            "period": period,
+            "height": parent.height + 1,
+            "proposer_address": proposer_address,
+            "proposer_bid": proposer_bid,
+        }
+        return cls(**header_kwargs)
+
     @property
     def hash(self):
-        header_hash = keccak(
-            b''.join((
+        """Hash for signing (excludes proposer signature)"""
+        return keccak(
+            b''.join([
                 int_to_bytes32(self.shard_id),
-                int_to_bytes32(self.expected_period_number),
-                self.period_start_prevhash,
                 self.parent_hash,
-                self.transaction_root,
-                pad32(self.coinbase),
-                self.state_root,
-                self.receipt_root,
-                int_to_bytes32(self.number),
-            ))
+                self.chunks_root,
+                int_to_bytes32(self.period),
+                int_to_bytes32(self.height),
+                pad32(self.proposer_address),
+                int_to_bytes32(self.proposer_bid),
+            ])
         )
-        return header_hash
+
+    def to_signed_collation(self):
+        raise NotImplementedError("TODO")
+
+
+class CollationHeader(rlp.Serializable):
+    """The header of a collation signed by the proposer."""
+
+    fields = [
+        ("shard_id", int32),
+        ("parent_hash", hash32),
+        ("chunks_root", hash32),
+        ("period", int32),
+        ("height", int32),
+        ("proposer_address", address),
+        ("proposer_bid", int32),
+        ("proposer_signature", Binary.fixed_length(96)),
+    ]
+
+    smc_field_sizes = [
+        ("shard_id", 32),
+        ("parent_hash", 32),
+        ("chunks_root", 32),
+        ("period", 32),
+        ("height", 32),
+        ("proposer_address", 32),
+        ("proposer_bid", 32),
+        ("proposer_signature", 96),
+    ]
+    smc_encoded_size = sum(size for _, size in smc_field_sizes)
+    assert [name for name, _ in smc_field_sizes] == [name for name, _ in fields]
+
+    def __repr__(self):
+        return "<CollationHeader #{0} {1} (shard #{2})>".format(
+            self.height,
+            encode_hex(self.hash)[2:10],
+            self.shard_id,
+        )
+
+    def to_bytes(self):
+        encoded = b''.join([
+            int_to_bytes32(self.shard_id),
+            self.parent_hash,
+            self.chunks_root,
+            int_to_bytes32(self.period),
+            int_to_bytes32(self.height),
+            pad32(self.proposer_address),
+            int_to_bytes32(self.proposer_bid),
+            self.proposer_signature,
+        ])
+        assert len(encoded) == self.smc_encoded_size
```

Comment:
We have tended to avoid assertions in runtime code as they are [easy to disable](https://stackoverflow.com/questions/1273211/disable-assertions-in-python).  Preference is to raise explicit exceptions like `ValueError` in cases like this.